src/qtism/data/content/BodyElement.php

```diff
@@ -272,4 +329,290 @@ public function getDir()
+    }
+
+    /**
+     * @param string $ariaLevel
+     * @throws InvalidArgumentException
+     */
+    public function setAriaLevel($ariaLevel)
+    {
+        if ($ariaLevel !== '' && !Format::isAriaLevel($ariaLevel)) {
+            $val = (is_object($ariaLevel)) ? ('instance of ' . get_class($ariaLevel)) : $ariaLevel;
+
+            $msg = "'${val}' is not a valid value for attribute 'aria-level'.";
+            throw new InvalidArgumentException($msg);
+        }
+
+        $this->ariaLevel = strval($ariaLevel);
+    }
+
+    /**
+     * @return string
+     */
+    public function getAriaLevel()
+    {
+        return $this->ariaLevel;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaLevel()
+    {
+        return $this->ariaLevel !== '';
+    }
+
+    /**
+     * @param int $ariaLive A value from the AriaLive enumeration or false for no value.
+     * @throws InvalidArgumentException
+     */
+    public function setAriaLive($ariaLive)
+    {
+        if ($ariaLive === false || in_array($ariaLive, AriaLive::asArray(), true)) {
+            $this->ariaLive = $ariaLive;
+        } else {
+            $val = (is_object($ariaLive)) ? ('instance of ' . get_class($ariaLive)) : $ariaLive;
+            $msg = "'${val}' is not a valid value for attribute 'aria-live'.";
+            throw new InvalidArgumentException($msg);
+        }
+    }
+
+    /**
+     * @return int
+     */
+    public function getAriaLive()
+    {
+        return $this->ariaLive;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaLive()
+    {
+        return $this->ariaLive !== false;
+    }
+
+    /**
+     * @param int $ariaOrientation A value from the AriaOrientation enumeration or false for no orientation.
+     * @throws InvalidArgumentException
+     */
+    public function setAriaOrientation($ariaOrientation)
+    {
+        if ($ariaOrientation === false || in_array($ariaOrientation, AriaOrientation::asArray(), true)) {
+            $this->ariaOrientation = $ariaOrientation;
+        } else {
+            $val = (is_object($ariaOrientation)) ? ('instance of ' . get_class($ariaOrientation)) : $ariaOrientation;
+            $msg = "'${val}' is not a valid value for attribute 'aria-orientation'.";
+            throw new InvalidArgumentException($msg);
+        }
+    }
+
+    /**
+     * @return false|int A value from the AriaOrientation enumeration or false if not orientation defined.
+     */
+    public function getAriaOrientation()
+    {
+        return $this->ariaOrientation;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaOrientation()
+    {
+        return $this->ariaOrientation !== false;
+    }
+
+    /**
+     * @param $ariaLabel
+     * @throws InvalidArgumentException
+     */
+    public function setAriaLabel($ariaLabel)
+    {
+        if (!is_string($ariaLabel)) {
+            $val = (is_object($ariaLabel)) ? ('instance of ' . get_class($ariaLabel)) : $ariaLabel;
```

Comment:
redundant, we already pretty sure that this is a string

Reply:
"pretty sure" yes indeed :smile: